lib/src/facts/aix/operating_system_resolver.cc

```diff
@@ -1,43 +1,31 @@
 #include <internal/facts/aix/operating_system_resolver.hpp>
+#include <internal/util/aix/odm.hpp>
+#include <facter/facts/collection.hpp>
+#include <facter/facts/fact.hpp>
 #include <facter/facts/os.hpp>
-#include <leatherman/execution/execution.hpp>
-#include <leatherman/file_util/file.hpp>
+#include <facter/facts/array_value.hpp>
+#include <facter/facts/map_value.hpp>
+#include <facter/facts/scalar_value.hpp>
 #include <leatherman/logging/logging.hpp>
-#include <leatherman/util/regex.hpp>
 
 #include <boost/algorithm/string.hpp>
+#include <odmi.h>
+#include <sys/cfgodm.h>
 
 using namespace std;
-using namespace leatherman::util;
-using namespace leatherman::file_util;
 using namespace boost;
-namespace execution = leatherman::execution;
+using namespace facter::util::aix;
 
 static string getattr(string object, string field)
 {
-    string result;
-
-    execution::each_line(
-        "/usr/sbin/lsattr", {"-El", object, "-a", field},
-        [&](string& line) {
-            if (!line.empty()) {
-                vector<string> tokens;
-                boost::split(tokens, line, boost::is_space(), boost::token_compress_on);
-                if (tokens.size() < 2) {
-                    return true;
-                }
-                result = tokens[1];
-                return false;
-            }
-            return true;
-        },
-        nullptr,
-        0);
-
-    if (result == "") {
-        LOG_WARNING("Could not get a value from lsattr -El {1} -a {2}", object, field);
+    string query = (boost::format("name = %1% AND attribute = %2%") % object % field).str();
+    auto attr_query = odm_class<CuAt>::open("CuAt").query(query);
+    for (auto& attr : attr_query) {
```

Comment:
It's a little confusing to use for loop here with an immediate return of the first element. I'd rather see the slightly more verbose manual iterator wrangling that makes it clear we're returning a single element.

```
auto attr = attr_query.begin()
if (attr != attr_query.end()) {
    return attr->value
}
```

Reply:
Hmm, it might be worthwhile to add a `value` helper to the `query_proxy` when we expect only a single value out of an ODM query?

Reply:
That seems like a better choice. I'll add the helper.

Reply:
Hmm, on second maybe it's best to have people use the more verbose iterator. B/c with the helper, we'd have to return a pointer to account for the possibility that the value may not exist. They'd still have to do an if check on that return and then dereference it.

Reply:
:+1: the iterator wrangling isn't that bad, if a helper is a hassle